test/Microsoft.Sbom.Targets.Tests/GenerateSbomE2ETests.cs

```diff
@@ -0,0 +1,458 @@
+        Assert.IsTrue(pack, "Failed to pack the project");
+    }
+
+    private void ExtractPackage()
+    {
+        // Unzip the contents of the NuGet package
+        var nupkgPath = Path.Combine(projectDirectory, "bin", configuration);
+        var nupkgFile = Path.Combine(nupkgPath, $"{expectedPackageName}.{expectedVersion}.nupkg");
+        var zipFile = Path.Combine(nupkgPath, $"{expectedPackageName}.{expectedVersion}.zip");
+        var extractPath = Path.Combine(projectDirectory, "bin", configuration, $"{Guid.NewGuid()}.temp");
+
+        // Rename the .nupkg file to .zip
+        File.Copy(nupkgFile, zipFile, true);
+
+        // Extract the .zip file
+        ZipFile.ExtractToDirectory(zipFile, extractPath);
+
+        manifestPath = Path.Combine(extractPath, "_manifest", sbomSpecificationDirectoryName, "manifest.spdx.json");
+        buildDropPath = extractPath;
+    }
+
+    [TestMethod]
+    public void SbomGenerationSucceedsForDefaultProperties()
+    {
+        //Arrange
+        // Create a Project object for ProjectSample1
+        var sampleProject = (Project)project;
+
+        // Get all the expected default properties
+        GetDefaultProperties(sampleProject);
+
+        // Set the TargetFrameworks property to empty. By default, it sets this property to net6.0 and net8.0, which fails for net8.0 builds.
+        sampleProject.SetProperty("TargetFrameworks", string.Empty);
+
+        // Restore, build, and pack the project
+        RestoreBuildPack(sampleProject);
+
+        // Extract the NuGet package
+        ExtractPackage();
+
+        // Validate the SBOM contents inside the package.
+        this.generatedSbomValidator.AssertSbomIsValid(manifestPath,
+            buildDropPath,
+            expectedPackageName,
+            expectedVersion,
+            expectedSupplier,
+            expectedNamespace,
+            null,
+            projectDirectory);
+    }
+
+    [TestMethod]
+    public void SbomGenerationSucceedsForValidManifestDirPath()
+    {
+        //Arrange
+        // Create a Project object for ProjectSample1
+        var sampleProject = (Project)project;
+
+        // Get all the expected default properties
+        GetDefaultProperties(sampleProject);
+
+        // Set the TargetFrameworks property to empty. By default, it sets this property to net6.0 and net8.0, which fails for net8.0 builds.
+        sampleProject.SetProperty("TargetFrameworks", string.Empty);
+
+        // Manually set the ManifestDirPath
+        sampleProject.SetProperty("SbomGenerationManifestDirPath", manifestDirPath);
+
+        // Restore, build, and pack the project
+        RestoreBuildPack(sampleProject);
+
+        // Extract the NuGet package
+        ExtractPackage();
+
+        manifestPath = Path.Combine(manifestDirPath, "_manifest", sbomSpecificationDirectoryName, "manifest.spdx.json");
+
+        // Check if the SBOM exists in the ManifestDirPath
+        Assert.IsTrue(File.Exists(manifestPath));
```

Comment:
I think this should test that: If we set the `manifestDirPath` to a relative path, the relative manifest path inside the Nuget package is the same and the content is valid.

We could also test what happens when we set `manifestDirPath` to an absolute path.

Reply:
I removed the ability to set ManifestDirPath altogether, as the desired location for the SBOM is at the root of the nuget package aka, BuildDropPath. For now I left the ManifestDirPath parameter empty in case people want to implement this in the future, but let me know if I should set this to something.